### PR TITLE
Fixes #4974 - Remove wantsauth dependency for setting FxA cookie

### LINKED ISSUE #4974: [FxA] Do not rely on wantsauth for FxA cookie-setting
Per some of the comments in #4966, we should be setting the accountid cookie in the `confirm-login` endpoint. We should also remove the special `?auth=1` handling, which is only necessary because it's triggering wantsauth.

### PR DESCRIPTION
**Pull request by punamdahiya**

### CODE REVIEW

File: server/src/reactruntime.js [hunk 1]
```diff
@@ -36,9 +36,8 @@ exports.HeadTemplate = class HeadTemplate extends React.Component {
     for (const locale of this.props.userLocales) {
       localeScripts.push(<script key={`l10n-${locale}`} src={this.props.staticLink(`/static/locales/${locale}.js`)} />);
     }
+    const wantsAuth = !this.props.authenticated;
```
Comment (punamdahiya) [inline server/src/reactruntime.js]: @chenba Including you this updated wantsAuth include criteria. With this change we will be loading wantsAuth on server only when deviceId doesn't exist

**Comment by punamdahiya** [comment]
Rest of the changes in PR are
a) Updating login and confirm login route handlers to use async -await
b)  create accountId cookie on successful firefox account authentication

I am keeping isOwner changes separate from this PR which need a new attribute added to shot and believe should be a separate PR.

File: server/src/reactruntime.js (hunk 1 above)
Comment (chenba) [reply server/src/reactruntime.js]: We also don't want `wantsAuth` when the user is already authenticated through FxA.

File: server/src/server.js [hunk 2]
```diff
@@ -605,6 +611,7 @@ app.post("/api/login", function(req, res) {
         return sendParams.accountId = accountId;
       }).then((accountId) => {
         if (vars.ownershipCheck) {
+
```
Comment (chenba) [inline server/src/server.js]: This feels unnecessary.  😃

File: server/src/server.js [hunk 3]
```diff
@@ -1091,10 +1098,13 @@ app.get("/api/fxa-oauth/login/*", function(req, res, next) {
...
-  }).catch(next);
+  } catch (err) {
+    mozlog.warn("fxa-oauth-login-failed", {err});
+    next(err);
+  }
 });
 
-app.get("/api/fxa-oauth/confirm-login", function(req, res, next) {
+app.get("/api/fxa-oauth/confirm-login", async function(req, res, next) {
   if (!req.deviceId) {
```
Comment (chenba) [inline server/src/server.js]: Wouldn't this `if` stop what we're  trying to do right away?

Comment (chenba) [reply server/src/server.js]: Oh, maybe I'm confusing this particular PR with the overall goal of having FxA work without device id?

Comment (punamdahiya) [reply server/src/server.js]: Yes,  this check will change to include accountId with PR that allows shots access and authentications on non-Firefox browser

**Pull request merged**